pkg/asyncdr/asyncdr.go

```diff
@@ -97,6 +97,119 @@ func CreateMigration(
 	return mig, err
 }
 
+func CreateMigrationSchedule(
+	name string,
+	namespace string,
+	clusterPair string,
+	migrationNamespace string,
+	includeVolumes *bool,
+	includeResources *bool,
+	startApplications *bool,
+	sp string,
+	suspend *bool,
+	autoSuspend bool,
+) (*storkapi.MigrationSchedule, error) {
+
+	migrationSpec := storkapi.MigrationSpec{
+		ClusterPair:       clusterPair,
+		IncludeVolumes:    includeVolumes,
+		IncludeResources:  includeResources,
+		StartApplications: startApplications,
+		Namespaces:        []string{migrationNamespace},
+	}
+	TemplateSpec := storkapi.MigrationTemplateSpec{
+		Spec: migrationSpec,
+	}
+
+	migrationSchedule := &storkapi.MigrationSchedule{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: storkapi.MigrationScheduleSpec{
+			Template:           TemplateSpec,
+			SchedulePolicyName: sp,
+			Suspend:            suspend,
+			AutoSuspend:        autoSuspend,
+		},
+	}
+	mig_sched, err := storkops.Instance().CreateMigrationSchedule(migrationSchedule)
+	return mig_sched, err
+}
+
+func CreateSchedulePolicy(policyName string, interval int) (pol *storkapi.SchedulePolicy, err error) {
+	schedPolicy, err := storkops.Instance().GetSchedulePolicy(policyName)
+	if err != nil {
+		log.InfoD("Creating a interval schedule policy %v with interval %v minutes", policyName, interval)
+		schedPolicy = &storkapi.SchedulePolicy{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name: policyName,
+			},
+			Policy: storkapi.SchedulePolicyItem{
+				Interval: &storkapi.IntervalPolicy{
+					IntervalMinutes: interval,
+				},
+			}}
+		schedPolicy, err = storkops.Instance().CreateSchedulePolicy(schedPolicy)
+	} else {
+		log.Infof("schedPolicy is %v already exists", schedPolicy.Name)
+	}
+	return schedPolicy, err
+}
+
+func WaitForNuOfMigration(sched_name string, sched_namespace string, count int, miginterval int) (map[string]string, error) {
+	migInterval := time.Minute * time.Duration(miginterval)
+	migTimeout := time.Minute * time.Duration(count*miginterval)
```

Comment:
The total number of migrations i.e `count` should not affect the time taken per migration. So this calculation would not be an accurate estimate of timeout per migration.

Reply:
Agree!, I will add a config for implementing this based on app time.